1.oid4vci_test_deployment.sh

```diff
@@ -210,6 +210,14 @@ $KC_INSTALL_DIR/bin/kcadm.sh create components -r master -o -f - < $WORK_DIR/sig
 echo "Creating OID4VCI client..."
 $KC_INSTALL_DIR/bin/kcadm.sh create clients -o -f - < $WORK_DIR/client-oid4vc.json || { echo 'OID4VCIClient creation failed' ; exit 1; }
 
+# Replace placeholder {{CLIENT_SECRET}} in JSON file with actual secret
+sed -i "s/{{CLIENT_SECRET}}/$CLIENT_SECRET/g" $WORK_DIR/openid4vc-rest-api.json 
```

Comment:
Sir, for that , when script 1 is ran, it get's the actual secret from the .env file  and replaces it in the json configuration file before exporting or deploying the client configuration on keycloak. I realized the script just exports  that configuarion as it's defined. So without that line, it will still create the client but the client secret will be "{{CLIENT_SECRET}}" and not the actual value

Reply:
I get that. But in the JSON configuration it is `${{CLIENT_SECRET}}` not `{{CLIENT_SECRET}}`. That is, your only replace `{{CLIENT_SECRET}}`, but `$` remains and ruins the secret. Second, when you have something you consider as a secret, you should never write it back to the file. You can read the JSON config in a variable, make the replacement in the variable (not the file), and reuse the variable on the next command.

Reply:
My bad, just saw that needed to removed "$", let me see how to address the changes

Reply:
With the jq tool, i was able to get the secret from the .env and pass to the json config file for the client without writing changes to disk